### PR TITLE
JMS service deletion

### PR DESCRIPTION
**Pull request by martaswiatkowska**
Service can be deleted only by JMS
Service deletion mean that Service status is set to "deleted"
Deleted service cannot be edited, published and offers cannot be edited, deleted, published/draft or added.
Deleted service isn't shown for users. In bacoffice are displayed with badge "deleted"

### CODE REVIEW

File: app/policies/backoffice/offer_policy.rb [hunk 1]
```diff
@@ -15,27 +15,28 @@ def resolve
...
   end
 
   def update?
-    managed?
+    managed? && !service_deleted?
   end
 
   def destroy?
-    managed? && orderless?
+    managed? && orderless? && !service_deleted? && !service_deleted?
```
Comment (goreck888) [inline app/policies/backoffice/offer_policy.rb:35]: Why did you add `!service_deleted?` twice?

File: app/policies/backoffice/service_policy.rb [hunk 2]
```diff
@@ -65,7 +65,9 @@ def create?
   end
 
   def update?
-    service_portfolio_manager? || (record.draft? && owned_service?)
+    (service_portfolio_manager? ||
+     (record.draft? && owned_service?)) &&
+    !record.deleted?
   end
 
   def publish?
```
Comment (goreck888) [inline app/policies/backoffice/service_policy.rb:73]: I think you should add `!record.deleted?` here and in `publish_unverified` too. I mean we shouldn't publish deleted services.

File: spec/policies/backoffice/offer_policy_spec.rb [hunk 3]
```diff
@@ -72,4 +72,22 @@
...
+  context "When offer service is deleted" do
+    let(:service) { create(:service, owners: [owner], status: :deleted) }
+    let(:offer) { build(:offer, service: service, status: :published) }
+
+    permissions :create?, :edit?, :update?, :destroy? do
+      it "danies access to service portfolio manager" do
+        expect(subject).to_not permit(service_portfolio_manager, offer)
+      end
+
+      it "danies access to service owner" do
```
Comment (goreck888) [inline spec/policies/backoffice/offer_policy_spec.rb:85]: Add tests for publish

File: app/policies/backoffice/service_policy.rb (hunk 2 above)
Comment (martaswiatkowska) [reply app/policies/backoffice/service_policy.rb:73]: ok but this status should be removed when rdt will be released I think

File: spec/policies/backoffice/offer_policy_spec.rb (hunk 3 above)
Comment (martaswiatkowska) [reply spec/policies/backoffice/offer_policy_spec.rb:103]: But offer cannot be published twice. Look at offer status

Comment (goreck888) [reply spec/policies/backoffice/offer_policy_spec.rb:103]: Ok, I didn't notice.
